src/pairs/Deck.py

```diff
@@ -42,6 +41,10 @@ def add_dict_discards(self, cards):
         for card_type, freq in cards.items():
             self.discards[card_type] += freq
 
+    def copy(self):
+        c = Deck(self.original_deck, self.curr_deck, self.discards, self.shuffling_limit)
```

Comment:
When you pass in `self.curr_deck`, since it's a dictionary, you're actually passing in the reference to the dictionary so when you mutate one of of the game states, you mutate the new one as well.

Also, I don't think this function belongs in this PR.

Reply:
This is still unaddressed. Make sure to pass in `self.curr_deck.deepcopy()` instead.

Reply:
addressed now